src/array_api_stubs/_draft/data_type_functions.py

```diff
@@ -50,7 +59,15 @@ def astype(
     Returns
     -------
     out: array
-        an array having the specified data type. The returned array must have the same shape as ``x``.
+        For ``dtype`` a data type, an array having the specified data type.
+        For ``dtype`` a kind of data type:
+
+        -   If ``x.dtype`` is already of that kind, the data type must be maintained.
+        -   Otherwise, ``x`` should be cast to a data type of that kind, according to the type promotion rules (see :ref:`type-promotion`) and the above notes.
+        -   Kinds must be interpreted as the lowest-precision standard data type of that kind for the purposes of type promotion. For example, ``astype(x, 'complex floating')`` will return an array with the data type ``complex64`` when ``x.dtype`` is ``float32``, since ``complex64`` is the result of promoting ``float32`` with the lowest-precision standard complex data type, ``complex64``.
```

Comment:
The text you've added here seems to only be thinking about the case where the `dtype` argument is a string, but the dtype being an actual dtype object is also still supported.

Reply:
Line 62 covers the dtype case. Do you have a suggestion to make it more clear?